### DISCUSSION TITLE
Non-convex Collision Box (Tank with Cannon)

### DISCUSSION THREAD

**Discussion by rumansaleem** [excaliburjs/Excalibur#1708] [Q&A]
Is there any work around for the proper collision detection for non convex bodies such as a tank with a cannon?
```
   +-+
   | |
+--+-+--+
|       |
|       |
|       |
+-------+

```

**Comment by eonarheim** [comment]
@rumansaleem I'll have some time to look at this sometime this week

**Comment by kamranayub** [comment]
At the moment I don't think you can do this with a single "collidable" body but specifically for this case, I would ask whether it's actually that important for the cannon to be collidable? In practice, for this I would use two Actors and move them in tandem. In every top-down tank game I've played, _usually_ the body moves independently from the cannon (e.g. keyboard moves tank, mouse cursor points cannon). This is also how a tank is physically put together so that would make sense. Unless I wanted things to collide with the cannon, I wouldn't make it collidable but you _could_ if they are two separate actors. That is how you'd handle it right now without official support for combining convex shapes into non-convex boundaries.

**Reply by rumansaleem** [reply]
Yes, keeping two different actors was what actually came to my mind but, it didn't feel quite right for cannon to have its own draw method with coordinates that are relative to tank and keeping all positions in sync manually.

But yeah thanks, this is the way to go I guess.

**Reply by kamranayub** [reply]
You're right, it does feel bad and there IS a feature that is available called Child Actors but the API is changing and there are known bugs with it. You could try it though, you can add any actor to another actor and the child will be positioned relative to the parent. 

https://excaliburjs.com/docs/api/edge/classes/_actor_.actor.html#add

Your mileage may vary though! 😬

**Reply by rumansaleem** [reply]
Yeah, I did try out the child actors, but then reverted to single actor after i read about it is being deprecated. 😅
